jetty-core/jetty-websocket/jetty-websocket-core-common/src/main/java/org/eclipse/jetty/websocket/core/WebSocketConnection.java

```diff
@@ -199,6 +199,43 @@ public void setUseOutputDirectByteBuffers(boolean useOutputDirectByteBuffers)
         this.useOutputDirectByteBuffers = useOutputDirectByteBuffers;
     }
 
+    /**
+     * Extra bytes from the initial HTTP upgrade that need to
+     * be processed by the websocket parser before starting
+     * to read bytes from the connection
+     *
+     * @param initialBuffer the bytes of extra content encountered during upgrade
+     */
+    protected void setInitialBuffer(ByteBuffer initialBuffer)
+    {
+        if (LOG.isDebugEnabled())
+            LOG.debug("Set initial buffer - {}", BufferUtil.toDetailString(initialBuffer));
+        try (AutoLock l = lock.lock())
+        {
+            networkBuffer = newNetworkBuffer(initialBuffer.remaining());
+        }
+        ByteBuffer buffer = networkBuffer.getByteBuffer();
+        BufferUtil.clearToFill(buffer);
+        BufferUtil.put(initialBuffer, buffer);
+        BufferUtil.flipToFlush(buffer, 0);
+    }
+
+    /**
+     * Physical connection Open.
```

Comment:
What does "Physical" mean here?

Reply:
These are pre-existing comments and I didn't write them.

I will remove these comments, I don't think they are more helpful than the inherited javadoc.